modules/objdetect/test/test_arucodetection.cpp

```diff
@@ -310,6 +310,148 @@ void CV_ArucoDetectionPerspective::run(int) {
 }
 
 
+/**
+ * @brief Draw 2D synthetic markers, temper with some pixels, detect them and compute their uncertainty.
+ */
+class CV_ArucoDetectionUnc : public cvtest::BaseTest {
+    public:
+    CV_ArucoDetectionUnc(ArucoAlgParams arucoAlgParam) : arucoAlgParams(arucoAlgParam) {}
+
+    protected:
+    void run(int);
+    ArucoAlgParams arucoAlgParams;
+};
```

Comment:
I think you can use existing tests for testing by adding a new parameter. Also we need tests with perspective-distorted markers.

Reply:
Yes, but the specificity of this test is that I tamper the cells of the markers (so I have a ground truth uncertainty) to show that the uncertainty works as expected. Using an other test, we will just have a super low uncertainty because no cells are tampered and this does not show that the approach works as expected.